### ISSUE TITLE
Packages attached via Depends are neglected

### ISSUE THREAD

**Issue by davidchall** [r-lib/downlit#56]
{downlit} adds links for functions in attached packages. It populates the list of attached packages by searching for strings like `library()` and `require()`. However, it's possible for a package to attach other packages via the `Depends` field of its DESCRIPTION file. Could we please consider adding these dependencies as attached packages too?

As an example, consider the [{gganimate} README](https://gganimate.com). It features a code block:

```r
library(ggplot2)
library(gganimate)

ggplot(mtcars, aes(factor(cyl), mpg)) +
  geom_boxplot() +
  # Here comes the gganimate code
  transition_states(
    gear,
    transition_length = 2,
    state_length = 1
  ) +
  enter_fade() +
  exit_shrink() +
  ease_aes('sine-in-out')
```

The {gganimate} package uses `Depends: ggplot2`, so calling `library(gganimate)` will automatically attach {ggplot2}. This issue is about making the `geom_boxplot()` link work without needing to call `library(ggplot2)` at the top.

**Comment by hadley** [comment]
I don't think this is a downlit problem, but a pkgdown problem — it should automatically add any packages in `Depends` to the `downlit.attached` option.
